jobbergate-composed/Dockerfile-slurm

```diff
@@ -0,0 +1,156 @@
+    && pushd http-parser \
+    && git checkout tags/v2.9.4 \
+    && make PREFIX=/usr \
+    && make install
+
+# JSON-C
+RUN set -x \
+    && git clone https://github.com/json-c/json-c.git \
+    && pushd json-c \
+    && git checkout tags/json-c-0.14-20200419 \
+    && mkdir build/ \
+    && cd build/ \
+    && cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release .. \
+    && make \
+    && make install
+
+# Use forked slurm with SKIP_LOCKDOWN
+RUN set -x \
+    && git clone https://github.com/omnivector-solutions/slurm.git \
+    && pushd slurm \
+    && git checkout dusktreader/hacked-slurm-for-docker \
+    && ./configure --enable-debug --prefix=/usr --sysconfdir=/etc/slurm \
+        --with-mysql_config=/usr/bin  --libdir=/usr/lib64 \
+        --enable-slurmrestd --with-libcurl \
+        --with-http-parser=/usr/local \
+        --with-yaml=/usr \
+        --with-json=/usr \
+        --with-jwt=/usr \
+    && make install \
+    && install -D -m644 etc/cgroup.conf.example /etc/slurm/cgroup.conf.example \
+    && install -D -m644 etc/slurm.conf.example /etc/slurm/slurm.conf.example \
+    && install -D -m644 etc/slurmdbd.conf.example /etc/slurm/slurmdbd.conf.example \
+    && install -D -m644 contribs/slurm_completion_help/slurm_completion.sh /etc/profile.d/slurm_completion.sh \
+    && popd \
+    && rm -rf slurm \
+    && groupadd -r --gid=995 slurm \
+    && useradd -r -g slurm --uid=995 slurm \
+    && mkdir /etc/sysconfig/slurm \
+        /var/spool/slurmd \
+        /var/run/slurmd \
+        /var/run/slurmdbd \
+        /var/lib/slurmd \
+        /var/log/slurm \
+        /data \
+    && touch /var/lib/slurmd/node_state \
+        /var/lib/slurmd/front_end_state \
+        /var/lib/slurmd/job_state \
+        /var/lib/slurmd/resv_state \
+        /var/lib/slurmd/trigger_state \
+        /var/lib/slurmd/assoc_mgr_state \
+        /var/lib/slurmd/assoc_usage \
+        /var/lib/slurmd/qos_usage \
+        /var/lib/slurmd/fed_mgr_state \
+    && chown -R slurm:slurm /var/*/slurm* \
+    && /sbin/create-munge-key
+
+# Install Singularity
+RUN yum install -y https://github.com/sylabs/singularity/releases/download/v3.10.2/singularity-ce-3.10.2-1.el7.x86_64.rpm
+
+# Install MPICH
+RUN yum install -y mpich-3.2 mpich-3.2-devel
+RUN echo "module load mpi/mpich-3.2-x86_64" >> /etc/bashrc
+
+ARG JWT_SECRET=supersecret
+
+COPY etc/slurmdbd.conf /etc/slurm/slurmdbd.conf
+RUN chown slurm:slurm /etc/slurm/slurmdbd.conf
+RUN chmod 600 /etc/slurm/slurmdbd.conf
+COPY etc/slurm.conf /etc/slurm/slurm.conf
+# JWKS doesn't seem to work. Makes slurmrestd throw a 500
+# COPY jwks.json /etc/slurm/jwks.json
+RUN echo -n "$JWT_SECRET" > /etc/slurm/jwt_hs256.key
+
+COPY etc/slurm-env.sh /etc/profile.d/slurm-env.sh
+
+# Add local-user
+RUN useradd -ms /bin/bash local-user
+
+COPY etc/slurm-entrypoint.sh /usr/local/bin/slurm-entrypoint.sh
+ENTRYPOINT ["/usr/local/bin/slurm-entrypoint.sh"]
+
+CMD ["slurmdbd"]
```

Comment:
this is nice stuff but too complex for this repository. I'd be more comfortable if we had a repository just to maintain this, then we can publish the image to the public and etc...

Reply:
I don't really agree. The slurm configuration and Dockerfiles found here are tailored very specifically for this use-case. It wouldn't be very useful in another context. It add anything by itself, and we'd constantly have to be pointing back to the Jobbergate main project. As it is, the users don't need to be concerned about the Docker and Slurm configuration if they are only interested in seeing how Jobbergate works.

I don't think it makes sense to publish the images either. Again, they are all targeted at this specific toy use-case. It would take A LOT more work and maintenance to make general-purpose images that would be appropriate to publish publicly, and I don't think we should tackle that problem at this point.

Having examples that can work in totality right out of the box is quite valuable.

Reply:
far enough mate. Resolved!